src/guides/v2.3/javascript-dev-guide/javascript/js_init.md

```diff
@@ -31,6 +31,65 @@ In Magento 2, you have two options for specifying declarative notation:
 
    > This is used to target either a CSS selector or `*`. If the CSS selector matches multiple HTML elements, the script will run for each matched HTML element. For `*`, no HTML element is selected and the script will run once with the HTML DOM as its target. This method can be implemented from anywhere in the codebase to target any HTML element. This is preferred when direct access to the HTML element is restricted, or when there is no target HTML element.
 
+Consider the example of adding a custom carousel JS
+
+-  Copy `<carousel_name>.carousel.js` to app/design/frontend/`<package_name>`/`<theme_name>`/web/js/`<carousel_name>`/
+
+-  Add your requirejs module at app/design/frontend/`<package_name>`/`<theme_name>`/web/js/carousel.js
+
+    ```javascript
+    define([
+    'jquery',
+    '<carousel_name>'
+    ], function($) {
+    return function(config, element) {
+        $(element).<carousel_name>(config);
+        };
+    });
+    ```
+
+-  Add requirejs config to app/design/frontend/`<package_name>`/`<theme_name>`/requirejs-config.js
+
+    ```javascript
+    var config = {
+    map:{
+        '*': {
+            'carousel': 'js/carousel',
+            '<carousel_name>': 'js/<carousel_name>/<carousel_name>.carousel'
+             }
+        }
+    };
+    ```
+ You now have two options for specifying declarative notation:
+
+-  use the `data-mage-init` attribute to insert the carousel in a certain element:
+
+    ```html
+    <div data-mage-init='{"carousel":{"option": value}}'>
+        <div class="item">Item 1</div>
+        ...
+        <div class="item">Item n</div>
+    </div>
+    ```
+
+-  use with `<script type="text/x-magento-init"/>`:
+
+    ```html
+    <div id="<carousel_name>" class="carousel">
+        <div class="item">Item 1</div>
+        ...
+        <div class="item">Item n</div>
+    </div>
+
+    <script type="text/x-magento-init">
+        {
+            "#<carousel_name>": {
+            "carousel": {"option": value}
+        }
```

Comment:
This block is not indented properly - braces should never be one on top of another